ISOProperties: Rename "Edit Config" and "Show Defaults"

Ebola16 opened the pull request:
Fixes https://bugs.dolphin-emu.org/issues/10775

These were very confusing to me due to the reasons outlined in the link above.

JosJuice commented on Source/Core/DolphinWX/ISOProperties/ISOProperties.cpp:
```diff
@@ -231,13 +231,16 @@ void CISOProperties::CreateGUIControls()
-  edit_config->SetToolTip(_("This will let you manually edit the INI config file."));
+  wxButton* const edit_config = new wxButton(this, ID_EDITCONFIG, _("Edit User Config"));
+  edit_config->SetToolTip(
+      _("This will let you edit the user config INI file(s) for this game. Changes should be "
```
`file` instead of `file(s)`. This only ever opens one file.

Ebola16 replied:
Done

JosJuice commented:
By the way, to explain why the text was the way it was: The default INIs are actually read-only sometimes, depending on the operating system and the install method. They're never read-only in Windows development builds, but they might be if you install a stable build (I haven't checked), and I think they're typically read-only on Linux. With that in mind, I think replacing the "read-only" part with a "please don't edit them" part is fine, since Windows development builds are a common use case after all.

MayImilae commented:
I'm fine with this, but... The Default SYS INIs should not be editable, on any operating system, without the user manually undoing the read-only block! I think we should look into fixing that before we adjust the naming of the setting.

Ebola16 commented:
Ok thanks, that makes more sense. I like my proposed changes too as it lets the user know that there are user and default configurations which was not clear before.

@MayImilae, I need to edit the default SYS INIs as some important options on Android aren't implemented in the UI yet. I understand wanting to prevent users from messing with those but it will make clean installs of Dolphin much more annoying to test.

MayImilae commented:
>I need to edit the default SYS INIs as some important options on Android aren't implemented in the UI yet.

But shouldn't that be addressed by allowing the creation of user INIs on android, not encouraging users edit the file with settings we know work and don't want them to touch?

Ebola16 commented:
Oh I see. A default ZIP of SYS INIs along with user-editable INIs wouldn't be a bad thing. I didn't think user SYS INI corruption was a big issue but I may be wrong on that.

Regardless, that probably shouldn't block this PR as the changed buttons don't open the SYS INIs and the phrasing works for both zipped and non-zipped default INIs. Let me know if I'm misunderstanding something as I didn't look at the zipped INI PR in great detail.

MayImilae commented:
True, it is worded well. It's vaguer than I'd like, but that kind of reflects the current vague status of it at the moment. I'll approve the change, in a totally just writing and not coding approval!

Still, I don't like the vagueness, and when situation changes the wording should change too. @delroth could you make sure that whenever #6178 is merged that the View Default Config button tooltip indicates that it is read only? Since it will properly be read only in your PR!

delroth commented:
Pretty sure I just removed the button in my PR anyway.

MayImilae commented:
@delroth Whoops you mentioned that in the PR. Well, works for me!

MayImilae commented on Source/Core/DolphinWX/ISOProperties/ISOProperties.cpp:
```diff
@@ -231,13 +231,16 @@ void CISOProperties::CreateGUIControls()
-  edit_config->SetToolTip(_("This will let you manually edit the INI config file."));
+  wxButton* const edit_config = new wxButton(this, ID_EDITCONFIG, _("Edit User Config"));
+  edit_config->SetToolTip(
+      _("This will let you edit the user config INI file for this game. Changes should be "
```
I know the old writing had a lot of this, but if we're touching it, let's make it better! First up, "you" is not great writing and in general should be avoided if possible. And Config should probably not be shortened in the tooltip. Also I think stating that editing should be here is a little redundant when it says not to do it on the other.

I recommend: "Allows manual editing of the user configuration INI file for this game."

MayImilae commented on Source/Core/DolphinWX/ISOProperties/ISOProperties.cpp:
```diff
@@ -231,13 +231,16 @@ void CISOProperties::CreateGUIControls()
-      _("Opens the default (read-only) configuration for this game in an external text editor."));
+      new wxButton(this, ID_SHOWDEFAULTCONFIG, _("View Default Config"));
+  edit_default_config->SetToolTip(_(
+      "This will let you view the default config INI file(s) for this game. Changes should be made "
```
Similar complaints here. You isn't great, and configuration shouldn't be shortened in the tooltip (at least not the first statement of the word). Also something a little more verbous about why they shouldn't edit it is good too (and helps further deter editing!).

I recommend: "Displays the default configuration INI file(s) for this game. These defaults are recommended settings from the developers to avoid known issues. Changes should be made to the user config INI files only, not to default config INI files."

Since delroth is just going to remove this button eventually anyway, I don't care toooo much about this, though.

ZexaronS replied:
What is the replacement to the button? Will it no longer be possible to display default INIs? That indeed wouldn't be needed if the whole content of the default INIs would be copied to the user INIs when the user decides to "restore defaults" for example.

ZexaronS commented:
I agree having the system default INIs treated as read-only (whether or not in ZIP), in terms that you can't edit them through the GUI, whether or not you have a filesystem read-only flag, either way anything shipped is editable if the user really wants to.

The user's INI settings would either be diffs to the defaults or a full user's copy which wouldn't be called a "user default" it would just be "user ini config" or something like that, it would just be a user setting and as long as the user uses his edits that would be saved into user config just as any other user config and would in practice work as a user default for a future session.

There could be a tickbox to switch between using user's INIs vs. system default INIs.

That said, if it was up to me, I would be in the favor of android having it's own android default INI defaults, so the user customization would be based on that, and not from desktop versions.

JosJuice commented:
> The user's INI settings would either be diffs to the defaults

That has already been the case for many years.

> That said, if it was up to me, I would be in the favor of android having it's own android default INI defaults, so the user customization would be based on that, and not from desktop versions.

I don't really think there are any situations that that would be useful for.

JMC47 commented:
I'd go a step further and say we're being negligent to have different GameINI defaults between Android and Desktop.

ZexaronS commented:
@JosJuice Someone mentioned differences on android platform, ... allright reading that again it seems like a minor temporary issue.

JosJuice commented:
> Someone mentioned differences on android platform, ... allright reading that again it seems like a minor temporary issue.

Ebola16 said some settings weren't implemented **in the UI**. What the UI supports has no effect on what the emulation core supports, so this is irrelevant for the default INIs.

Ebola16 commented:
Yeah @delroth removed the "Show defaults" button. Do we really want to remove it though? It's an easy way for users to check that they aren't overriding something that they shouldn't and you don't need to go searching through a massive zip in the future. I don't feel too strongly for keeping it as I've never used the button but I do see a use case for it.

Ebola16 commented:
After actually reading the comments of #6178 I now think "Show Defaults" should be dropped eventually.

This PR will probably be ready to go before #6178 and is better than the current button names so this should probably just be merged after review. @delroth can then remove the newly-named "View Default Config" button again.

@MayImilae I incorporated your suggestions with the intention of losing "View Default Config" eventually. I added the "Settings in the user config INI override default config INI settings." sentence to actually answer the question of how one should properly edit values set in the default config INI.

@ZexaronS I think @JosJuice addressed most points but I wanted to add that anything that involved users disabling all user configs shouldn't even be considered until the default configs are read-only.
